### PR TITLE
Source Google Search Console: add custom analytics stream

### LINKED ISSUE #13034: Source Google Search Console: Add Search_analytics_custom stream

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
*Add custom stream support. Now its possible to specify stream with custom defined dimensions*

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

**Comment by bazarnov** [comment]
@roman-yermilov-gl 
Please fix `FlakeCheck` warnings: https://github.com/airbytehq/airbyte/runs/8251151667?check_suite_focus=true#step:11:540

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py [hunk 1]
```diff
@@ -62,8 +63,21 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
...
 
+        streams = streams + self.get_custom_reports(config=config, stream_config=stream_config)
+
         return streams
 
+    def get_custom_reports(self, config: Mapping[str, Any], stream_config: Mapping[str, Any]) -> List[Optional[Stream]]:
+        if "custom_reports" not in config:
+            return []
+
+        reports = json.loads(config["custom_reports"])
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py:74]: For safety would you mind checking the `custom_reports` field is a list?

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json [hunk 2]
```diff
@@ -105,6 +105,12 @@
             }
           }
         ]
+      },
+      "custom_reports": {
+        "order": 4,
+        "type": "string",
+        "title": "Custom Reports (Optional)",
+        "description": "A JSON array describing the custom reports you want to sync from Google Search Console. See <a href=\"https://docs.airbyte.com/integrations/sources/google-search-console#step-2-set-up-the-google-search-console-connector-in-airbyte\">the docs</a> for more information about the exact format you can use to fill out this field."
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json:113]: Could we validate this JSON string's structure to ensure it's a valid custom report declaration?

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json [hunk 3]
```diff
@@ -105,6 +105,12 @@
             }
           }
         ]
+      },
+      "custom_reports": {
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json:109]: Please update our internal secrets config for this connector with this field set to ensure SAT run is testing this stream with data.

**Review by alafanechere — CHANGES_REQUESTED** [review]
Thank you for this addition. I request a change because the SAT is not testing this stream with our current secret: you need to define the `custom_reports` field in our GSM secrets.

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py (hunk 1 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/source.py:101]: Done by defining jsonschema

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json (hunk 3 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json:109]: Done by defining jsonschema

Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json:109]: Updated

File: airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json (hunk 2 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-google-search-console/source_google_search_console/spec.json:113]: Done by defining jsonschema

**Review by alafanechere — CHANGES_REQUESTED** [review]
Thanks @roman-yermilov-gl for the changes. Could you also add these new streams to the catalogs that are used for integration tests? The new stream is still not yet covered by the acceptance tests.

**Comment by roman-yermilov-gl** [comment]
> 



> Thanks @roman-yermilov-gl for the changes. Could you also add these new streams to the catalogs that are used for integration tests? The new stream is still not yet covered by the acceptance tests.

Done

**Review by alafanechere — APPROVED** [review]
Thank you for the catalog changes!

**Review by arsenlosenko — APPROVED** [review]

**Pull request merged**